### DISCUSSION TITLE
AS3 matrix

### DISCUSSION THREAD

**Discussion by huwdp** [ruffle-rs/ruffle#1539] [Ideas]
I have created this page here (https://huwdp.github.io/Flash-Matrix/) a while ago to track AS3 support between the different Flash implementations and was wondering how I could extract Ruffle AS3 API support in the future?

I think it would be good to see the progress of Ruffle and to help others easily see what needs to be implemented in Ruffle.

**Comment by ROBERT-MCDOWELL** [comment]
excellent page indeed, it would be very nice to include also AVM1 as2 in your table.

**Reply by huwdp** [reply]
Hi there, I will look into this. Need to figure out a way of generating the table for AVM1

**Comment by Herschel** [comment]
For AS1/2, we generated [tracking issues](https://github.com/ruffle-rs/ruffle/issues?q=is%3Aissue+is%3Aopen+%22Actionscript+Tracking%22) to keep track of which APIs were implemented. We may do the same for AS3, although the number of APIs is much larger, so perhaps there is a better idea.

What sort of data format do you use to generate the matrix? Possibly we could use this format could to track our progress internally, as well.

**Reply by huwdp** [reply]
Hi there, I use scrapper.py in the repo [here](https://github.com/huwdp/Flash-Matrix) (https://github.com/huwdp/Flash-Matrix) to generate the list of constants, methods and properties implemented in Flash.

Then for the matrix itself I run buildtable.py. This script looks for Not Implemented Logging function calls and extracts the APIs noted in the arguments. Not perfect but it'll do for now.

**Comment by ghost** [comment]
Having this in the wiki would be awesome :) @huwdp @Herschel

**Comment by huwdp** [comment]
Ruffle has been added to the support matrix.
